include/server/mir/scene/surface.h

```diff
@@ -142,6 +142,15 @@ class Surface :
 
     /// The session this surface was created by
     virtual auto session() const -> std::weak_ptr<Session> = 0;
+
+    /// Sets the geometry of the margins around a surface
+    /// Margins make room for server-side-decorations, which are attached as a child surface
+    /// This call will trigger the content_size to be adjusted and the window_size will remain unchanged
+    virtual void set_window_margins(
+        geometry::DeltaY top,
+        geometry::DeltaX left,
+        geometry::DeltaY bottom,
+        geometry::DeltaX right) = 0;
```

Comment:
I'm tempted by:
```c++
    virtual void set_window_geometry(
        geometry::DeltaY top_margin,
        geometry::DeltaX left_margin,
        geometry::DeltaY bottom_margin,
        geometry::DeltaX right_margin) = 0;
```
but I'm not going to block on it. (Just mention it and see what happens.)

Reply:
No, on account of there being a confusingly similar but distinct Wayland message called `set_window_geometry`.